package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,345 @@
+       ``[frame, residue]``, where `structure_type` is one of three characters:
+       ['H', 'E', '-'], representing alpha-helix, sheet and loop, respectively.
+
+   .. attribute:: results.dssp_ndarray
+
+       Contains the one-hot encoding of the time series of the DSSP assignment
+       as chars N×mx3 :class:`numpy.ndarray` array with content
+       ``[frame, residue, encoding]``, where ``encoding`` is a (3,) shape
+       :class:`numpy.ndarray` of booleans with axes representing loop '-',
+       helix 'H' and sheet 'E', consequently.
+"""
+
+import numpy as np
+from MDAnalysis import Universe
+
+from ..base import AnalysisBase
+from ...due import due, Doi
+
+due.cite(
+    Doi("10.1002/bip.360221211"),
+    description="DSSP algorithm description",
+    path="MDAnalysis.analysis.dssp",
+    cite_module=True,
+)
+
+del Doi
+
+
+try:  # pragma: no cover
+    from pydssp.pydssp_numpy import (
+        assign,
+        _get_hydrogen_atom_position,
+    )
+
+    HAS_PYDSSP = True
+
+except ModuleNotFoundError:
+    HAS_PYDSSP = False
+    from .pydssp_numpy import (
+        assign,
+        _get_hydrogen_atom_position,
+    )
+
+
+def translate(onehot: np.ndarray) -> np.ndarray:
+    """Translate a one-hot encoding summary into char-based secondary
+    structure assignment. One-hot encoding corresponds to C3 notation:
+    '-', 'H', 'E' are loop, helix and sheet, respectively. Input array must
+    have its last axis of shape 3: (n_residues, 3) or (n_frames, n_residues, 3)
+
+    Examples
+    --------
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> import numpy as np
+    >>> # encoding 'HE-'
+    >>> onehot = np.array([
+        [False, True, False],  # 'H'
+        [False, False, True],  # 'E'
+        [True, False, False]]) # '-'
+    >>> ''.join(translate(onehot))
+    'HE-'
+
+    Parameters
+    ----------
+    onehot : np.ndarray
+        input array of one-hot encoding in ('-', 'H', 'E') order
+
+    Returns
+    -------
+    np.ndarray
+        array of '-', 'H' and 'E' symbols with secondary structure
+
+    .. versionadded:: 2.8.0
+    """
+    C3_ALPHABET = np.array(["-", "H", "E"])
+    index = np.argmax(onehot, axis=-1)
+    return C3_ALPHABET[index]
+
+
+class DSSP(AnalysisBase):
+    """Assign secondary structure using DSSP algorithm.
+
+    Analyze a selection containing a protein and assign secondary structure
+    using the Kabsch-Sander algorithm [Kabsch1983]. Only a subset of secondary
+    structure categories are implemented:
+
+    - 'H' represents a generic helix (α-helix, π-helix or 3-<sub>10</sub> helix)
+    - 'E' represents 'extended strand', participating in beta-ladder (parallel
+      or antiparallel)
+    - '-' represents unordered part
+
+    The implementation was taken from the pydssp package (v. 0.9.0)
+    https://github.com/ShintaroMinami/PyDSSP by Shintaro Minami.
+
+    Parameters
+    ----------
+    u : Universe
+        input universe
```

Comment:
Are you able to also use an AtomGroup? Is there anything in the code that _requires_ a Universe?